accepted/2021/System.Diagnostics/Metrics-Design.md

```diff
@@ -0,0 +1,502 @@
+# Metrics APIs Design
+
+## Overview
+
+This document is discussing the .NET Metrics APIs design which implements the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/new_api.md).
+
+The OpenTelemetry Metrics APIs support reporting measurements about the execution of a computer program at run time. The Metrics APIs are designed explicitly for processing raw measurements, generally with the intent to produce continuous summaries of those measurements, efficiently and simultaneously.
+
+The OpenTelemetry architecture is splitting the measurement reporting part to a component called `APIs` and that is the part we are discussing here and proposing APIs for it. The other part is called `SDK` which will monitor, aggregate the Metrics measurements, and export the aggregated measurement to some backend. The SDK part will be implemented in the OpenTelemetry [dotnet repo](https://github.com/open-telemetry/opentelemetry-dotnet) and will not part of the .NET. The `SDK` will depend on the APIs we are going to expose in .NET.
+
+The proposed APIs will be used by the application or Library authoress to record metrics measurements. The APIs also include a listener which will allow listening to the recorded measurements events which can be used by the OpenTelemetry SDK to aggregate the measurements and export them.
+
+The proposed APIs are intended to be supported on Full Framework 4.6 and up, .NET Core all versions, and .NET 5.0 and up.
```

Comment:
.NET Core 2.1/3.1 and above?

Reply:
I think we can reword it as all supported .NET Core versions instead.

Reply:
yea. I want to specifically ask about .NET Core 2.1 which will be out-of-support in August, before .NET 6.
So the support would be:
.NET Framework 4.6 and above
.NET Core 3.1
.NET 5 and above

Reply:
Officially we'll support the versions that are not out of support yet. But, Systsem.Diagnostics.Diagnostics source support netstandard2.0 which will make it work with net core 2.1.

Reply:
That remind - could you add in the intro that this new API will be part of DiagnosticSource package itself.